### PR TITLE
sympy: 1.6.2 -> 1.7.1

### PR DESCRIPTION
**Pull request by suhr**
###### Motivation for this change

Required by Mathics (https://github.com/NixOS/nixpkgs/pull/107652).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by timokau** [comment]
Have you tested if `sage` still builds after this update?

**Comment by suhr** [comment]
@timokau No. I thought the CI does that?

**Comment by timokau** [comment]
No, ofBorg only builds the package itself by default. Building all reverse-dependencies of every PR would take too much resources. Its possible to tell ofBorg to build additional packages, but in this case it would most likely time out.

**Comment by collares** [comment]
According to https://trac.sagemath.org/ticket/30985, you probably need to apply a small patch available at https://git.sagemath.org/sage.git commit?id=bb9ceadc8a56a3fd5cb3b737d22ac9bfa4a9cbac by adding it to https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/math/sage/sage-src.nix#L95. Let me know if you encounter any issues.

**Comment by collares** [comment]
Note that the `singular.pyx` Sage segfault is intermittent, doesn't happen too often, happens upstream too and is unfixed there (https://trac.sagemath.org/ticket/30945). If that's the only failure (which was the case on the x86_64 builder), I think we can consider the test run a success.

**Comment by collares** [comment]
Yep, this new run was all green, both on aarch64 and on x86_64. Sage seems OK with the update.

File: pkgs/development/python-modules/sympy/default.nix [hunk 1]
```diff
@@ -8,11 +8,11 @@
 
 buildPythonPackage rec {
   pname = "sympy";
-  version = "1.6.2";
+  version = "1.7.1";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1cfadcc80506e4b793f5b088558ca1fcbeaec24cd6fc86f1fdccaa3ee1d48708";
+    sha256 = "sha256-o96SYel1Nbg7uGB7DaLH0DEmZQ+v6isniWV7Ipwkay4=";
```
Comment (omasanori) [inline pkgs/development/python-modules/sympy/default.nix:15]: If the SRI hash format is not allowed, you can use this instead:
```suggestion
    sha256 = "0bkb4jf24yv5i4kjpsmg1xjjccfhqyi0syv0p0xvhdbmx5hr5pm3";
```
(converted by `nix to-base32 sha256-o96SYel1Nbg7uGB7DaLH0DEmZQ+v6isniWV7Ipwkay4=`)

**Review by omasanori — COMMENTED** [review]
Just to confirm, is the SRI hash allowed to use in nixpkgs? If so, I have no objection.
Thank you all for your effort!

File: pkgs/development/python-modules/sympy/default.nix (hunk 1 above)
Comment (suhr) [reply pkgs/development/python-modules/sympy/default.nix:15]: There's a lot of packages with SRI hash in nixpkgs, so I believe it's allowed.

Comment (omasanori) [reply pkgs/development/python-modules/sympy/default.nix:15]: OK. Thank you for confirmation.

**Comment by timokau** [comment]
> Note that the `singular.pyx` Sage segfault is intermittent

In that case I think it would be best to disable that test. Otherwise users will have to run the whole testsuite locally whenever the test fails on hydra.

That is unrelated to this PR though. Thanks for the reviews @collares and @omasanori!

**Review by timokau — APPROVED** [review]

**Comment by timokau** [comment]
Took the liberty to squash & merge to avoid an intermediate commit where sage is broken.

**Pull request merged**